components/feature/media/src/main/java/mozilla/components/feature/media/state/MediaStateMachine.kt

```diff
@@ -118,9 +132,19 @@ private class MediaSessionObserver(
         updateState()
     }
 
+    @Synchronized
     private fun updateState() {
-        val state = determineNewState()
-        stateMachine.transitionTo(state)
+        // We delay updating the state here and cancel previous jobs in order to batch multiple
+        // state changes together before updating the state. In addition to reducing the work load
+        // this also allows us to treat multiple pause events as a single pause event that we can
+        // resume from.
+        updateStateJob?.cancel()
+        updateStateJob = GlobalScope.launch {
```

Comment:
Is there any specific reasons to use`GlobalScope`?

Reply:
Mostly because MediaStateMachine is a singleton object and doesn't have a lifetime. But then again it has start/stop methods. So I could create a scope in start() and cancel it in stop(). I'll update the PR.